glean-core/src/error_recording.rs

```diff
@@ -69,7 +69,6 @@ fn get_error_metric_for_metric(meta: &CommonMetricData, error: ErrorType) -> Cou
     // Can't use meta.identifier here, since that might cause infinite recursion
     // if the label on this metric needs to report an error.
     let identifier = meta.base_identifier();
-    let name = strip_label(&identifier);
```

Comment:
But isn't this potentially be called on a metric that was created as a labeled metric, and thus the name will include its label?

Reply:
How is that possible? Do we use this pattern of creating a metric as labeled outside of the error_recording module? 

Users can't define metrics like that in the YAML files, glean_parser would complain about that.

I mean, I am ok with dropping this change. It's just confusing that we use base_indentifier, which gets the identifier without the label and immediatelly follow that with strip_label.

Reply:
When getting a metric from a label on a labeled metric[^1], we use `name/label` as the internal name of that newly constructed metric. It's also how we store the data.
Errors on that (sub)metric are reported on the user-defined labeled metric, that's why we strip out the label during error recording.
See e.g. https://github.com/mozilla/glean/blob/dfd41e42c5d89d587d015cf6b328bcee00e900b5/glean-core/src/metrics/labeled.rs#L96-L99

Now the fact that no test failed means we don't actually have any test coverage for such errors.


[^1]: What a sentence ....

Reply:
TIL! Thanks for the detailed explanation. This is great to know because I would have introduced a bug in Glean.js if I didn't add this, heh.

I dropped the first commit and added a test that fails with the previous change.